geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureUtils.scala

```diff
@@ -87,6 +90,7 @@ object AvroSimpleFeatureUtils {
       case c if classOf[Geometry].isAssignableFrom(c)            => baseType.bytesType.noDefault
       case c if classOf[java.util.List[_]].isAssignableFrom(c)   => baseType.bytesType.noDefault
       case c if classOf[java.util.Map[_, _]].isAssignableFrom(c) => baseType.bytesType.noDefault
+      case c if classOf[Array[Byte]].isAssignableFrom(c)         => baseType.bytesType.noDefault
```

Comment:
are there any potential type-erasure issues here? like it might match on an array of something other than bytes. Not sure if there is any way to avoid that though...